core/src/main/scala/org/apache/spark/sql/streaming/DirectKafkaStreamSource.scala

```diff
@@ -65,8 +68,13 @@ final class DirectKafkaStreamRelation(
   private val startingOffsets = getStartingOffsets
 
   private def getStartingOffsets = {
-    DirectKafkaStreamRelation.partitionOffsetMethod.invoke(null, options("startingOffsets"))
-        .asInstanceOf[Map[TopicPartition, Long]]
+
+    if (options.contains(STARTING_OFFSETS_PROP)) {
+      partitionOffsetMethod.invoke(null, options(STARTING_OFFSETS_PROP))
+          .asInstanceOf[Map[TopicPartition, Long]]
+    } else {
```

Comment:
I prefer using a single lookup in maps instead of contains+apply (esp since this is case-insensitive map so has even more operations). Though perf does not matter here, but a good practise nevertheless leading to cleaner pattern matching and to avoid cases where one may miss that it can have some perf impact:

```scala
options.get(STARTING_OFFSETS_PROP) match {
  case Some(offsets) => ...
  case None => ...
}
```